### ISSUE TITLE
Spellcheck should be initiated only after a word has been completed

### ISSUE THREAD

**Issue by joelodge** [typora/typora-issues#983]
When the user is typing (and not in source code mode), the spellcheck feature analyses and frequently underlines partially complete words, resulting in flickering red underlining which only disappears once a word is complete.

This is distracting and makes for a less pleasant typing experience. Two potential fixes come to mind:

1. Spellcheck could only analyse a word after it has been completed (i.e. followed with a space character or punctuation other than an apostrophe or hyphen) - I would suggest this is the most effective option.
2. Spellcheck could be subject to a short (~500 ms) delay

**Comment by abnerlee** [comment]
merge into https://github.com/typora/typora-issues/issues/214
